test-suite/src/main/scala/testsuite/core/IsInstanceOfTest.scala

```diff
@@ -50,6 +52,51 @@ object IsInstanceOfTest {
     !base2.isInstanceOf[Base]
   }
 
+  private def testArrays(): Unit = {
+    val child: Any = new Child
+    val booleanArray: Any = new Array[Boolean](1)
+    val intArray: Any = new Array[Int](1)
+    val stringArray: Any = new Array[String](1)
+    val intArrayArray: Any = new Array[Array[Int]](1)
+    val stringArrayArray: Any = new Array[Array[String]](1)
+    val objectArray: Any = new Array[Object](1)
+
+    // direct instances
+    ok(booleanArray.isInstanceOf[Array[Boolean]])
+    ok(intArray.isInstanceOf[Array[Int]])
+    ok(stringArray.isInstanceOf[Array[String]])
+    ok(intArrayArray.isInstanceOf[Array[Array[Int]]])
+    ok(stringArrayArray.isInstanceOf[Array[Array[String]]])
+    ok(objectArray.isInstanceOf[Array[Object]])
+
+    // primitive arrays are not instances of any other type
+    ok(!booleanArray.isInstanceOf[Array[Int]])
+    ok(!booleanArray.isInstanceOf[Child])
+    ok(!booleanArray.isInstanceOf[Array[Array[Boolean]]])
+    ok(!booleanArray.isInstanceOf[Array[Object]])
+
+    // different dimensions
+    ok(!intArray.isInstanceOf[Array[Array[Int]]])
+    ok(!intArrayArray.isInstanceOf[Array[Int]])
+    // TODO ok(!intArrayArray.isInstanceOf[Array[Array[Array[Int]]]])
+    // TODO ok(!stringArrayArray.isInstanceOf[Array[String]])
+    // TODO ok(!stringArrayArray.isInstanceOf[Array[Array[Array[String]]]])
+    // TODO ok(!objectArrayArray.isInstanceOf[Array[Array[Object]]])
+
+    // reference array types are covariant
+    ok(stringArray.isInstanceOf[Array[Object]])
+    ok(intArrayArray.isInstanceOf[Array[Object]])
+    // TODO ok(!objectArray.isInstanceOf[Array[String]])
```

Comment:
[note]
this is the case we have a problem that is mentioned in `genIsInstanceOf`https://github.com/tanishiking/scala-wasm/pull/41/files#diff-6c1484da3abfe8758c135ddd2cb90e4805486674a0d9ef0f27b2253280966408R1118 ``